pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -134,7 +96,36 @@ func (nc *NamespaceController) Run(stopCh <-chan struct{}) {
 		log.Error("Failed to sync namespace controller cache")
 	}
 	log.Infof("Namespace controller started")
-	go nc.queue.Run(stopCh)
+
+	go wait.Until(nc.runWorker, time.Second, stopCh)
+	go func() {
```

Comment:
no need to run this in a goroutine. I realize the old code did, but its already called in a goroutine

Reply:
I see that the `Run` is used as a non-blocked func in the `TestNamespaceController`.

Reply:
fixed